src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -123,8 +147,40 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
 
+     async verifyCard(methodId: string , amount: number,  payment: any): Promise<GooglePayVerifyPayload>  {
+        if (methodId === PaymentStrategyType.BRAINTREE_GOOGLE_PAY) {
+            const { nonce } = payment.paymentData;
+            const threeDSecure = await this._braintreeSDKCreator?.get3DS();
+
```

Comment:
Maybe its better to check nonce and threeDSecure here, before passing values to the `_braintreePresent3DSChallenge` method

```suggestion
        if (!nonce || !threeDSecure) {
            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
        }
```

Reply:
fixed